### PR TITLE
Replace Button with BrowserButton, set common font properties for BrowserButton, etc.

### LINKED ISSUE #9223: button style inconsistency on about:preferences (button normalization)
**Describe the issue you encountered:** the button style on `about:preferences#sync` is different with that on the other tabs of `about:preferences` such as `about:preferences#payments` 

screenshot 2017-06-03 14 17 26 screenshot 2017-06-03 14 17 34

The button on the left looks thinner and the font-size looks smaller than the button on the right.

See: https://github.com/brave/browser-laptop/blob/548e11b1c889332fadb379237555625ad2a3c845/less/button.less#L160

Also the styles of buttons on browser's UI such as bookmark hanger dialog is different with them.

screenshot 2017-06-03 14 19 24

See: https://github.com/brave/browser-laptop/blob/734cefcc993fdd73c7b8266d90a8dd7aba9fbcdf/less/window.less#L9

You would see the different font-family is specified to each of them.

We are on the way of normalizing the buttons by converting `buttton` component with `browserButton` component. We would have to investigate which `font-size`, `font-family`, and `line-height` (the properties which can/must be specified with `font` property) should be applied as to `browserButton` inside `browserButton.js`.

- Brave Version (revision SHA): 0.15.314

- Extra QA steps:
    1.
    2.
    3.

- Any related issues:

### LINKED ISSUE #10815: Button text looks blurry on Windows
- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
Button text looks blurry on Windows 

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows 10 x64

- Brave Version (revision SHA):

Brave | 0.18.27
-- | --
rev | eba55a1
Muon | 4.3.16

- Steps to reproduce
    1. Check for update, see button text, looks blurry
    2. Hover over the button, both button and text looks sharp

- Actual result:
See screenshot

- Expected result:
Text should be sharp irrespective of hover only button should show the hover effect

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
Yes verified on older builds same issue

- Can this issue be consistently reproduced?
Yes

- Extra QA steps:
    1. Verified on 100% DPI same issue but not very noticeable as everything looks very small
    2. Adding `-webkit-font-smoothing: antialiased;` makes no difference
    3.

- Screenshot if needed:
125% DPI
buttontext
100%DPI
buttontext1

- Any related issues:
cc: @luixxiul @cezaraugusto @jenn-rhim

### LINKED ISSUE #10946: Add browserButton_smallItem
**Describe the issue you encountered:**
Add `browserButton_smallItem` for delete confirmation button on `downloadItem.js`

screenshot 2017-09-14 17 08 00

- Extra QA steps:
    1.
    2.
    3.

- Any related issues:

### LINKED ISSUE #11114: Add cursor: 'pointer' to browserButton by default
### Description

Add `cursor: 'pointer'` to browserButton by default to avoid duping

### Additional Information

Related: #11111

### LINKED ISSUE #11115: Replace span with onClick with browserButton
### Description

Replace span with onClick with browserButton.

### PR DESCRIPTION
**Pull request by luixxiul**
**Requires https://github.com/brave/browser-laptop/pull/11547**
**Blocks #10881**

Fixes #10815
Closes #9223
Closes #11114
Closes #11115
Addresses #7168
Addresses #10898

# Test Plans

## 1st commit

Addresses #7168

Test Plan 1:
1. Open about:autofill
2. Click 'Add Address' and 'Add Credit Card'
3. Make sure the buttons are displayed

Test Plan 2:
1. Open about:preferences#payments
2. Click the setting icon
3. Make sure the close button on the modal dialog is displayed

Test Plan 3:
1. Open https://webtorrent.io/free-torrents
2. Click any link to a torrent file
3. Click 'Start Download'
4. Make sure the button is changed to 'Stop Download'

Test Plan 4:
1. Open clear browsing data panel
2. Make sure there is margin between the buttons
3. Open import browser data panel
4. Repeat 2
5. Open http://browserspy.dk/password.php
6. Click a link to password-dk.php
7. Repeat 2
8. Open https://mixed-script.badssl.com/
9. Click the lock icon
10. Repeat 2
11. Disable Widevine
11. Open https://netflix.com to display the widevine panel
12. Repeat 2

## 2nd commit

Fixes #10815
Closes #9223 

Test Plan:
1. Open about:preferences and display a button
2. Open data import dialog
3. Make sure the browser label is rendered in the same font properties

screenshot 2017-09-22 0 34 33 screenshot 2017-09-22 0 35 02

## 3rd commit

Closes #11114
Closes #11115
Addresses #10898

Test Plan:
1. Open about:styles
2. Click 'buttons'
3. Make sure cursor becomes pointer on hover
4. Open about:preferences#payments
5. Click the advanced setting
6. Make sure the close button of the modal overlay becomes pointer

Test Plan 2:
1. Open brave.com
2. Click the shield button
3. Make sure that the close button, info button and reload button has pointer cursor on hover

Test Plan 3:
1. Open about:preferences#plugins
2. Make sure the info buttons under Widevine support has pointer cursor on hover

Test Plan 4:
1. Open about:brave
2. Click copy button
3. Make sure the version information is copied

----

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/renderer/components/main/siteInfo.js [hunk 1]
```diff
@@ -68,8 +68,7 @@ class SiteInfo extends React.Component {
         <span className={cx({
           fa: true,
           'fa-lock': true,
-          [css(styles.secureIcon__fa)]: true,
-          [css(styles.secureIcon__extendedValidation)]: this.props.isExtendedValidation
+          [css(styles.secureIcon__fa, this.props.isExtendedValidation && styles.secureIcon__extendedValidation)]: true
```
Comment (luixxiul) [inline app/renderer/components/main/siteInfo.js]: This doesn't follow the BEM strictly; I'll work on the file including this line with another PR.

File: app/renderer/components/download/downloadItem.js [hunk 2]
```diff
@@ -149,96 +152,117 @@ class DownloadItem extends React.Component {
...
-        }
+      <div className={cx({
+        downloadActions: true,
+        [css(styles.downloadItems__item__actions)]: true
+      })}>
+        <div className={css(styles.downloadItems__item__actions__left)}>
+          {
+            this.props.allowPause
+            ? <BrowserButton
+              iconOnly
```
Comment (luixxiul) [inline app/renderer/components/download/downloadItem.js]: `iconOnly` rocks 🎸

**Comment by luixxiul** [comment]
Requesting a review from @srirambv on his Windows environment.

File: app/renderer/components/common/button.js [hunk 3]
```diff
@@ -1,51 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (cezaraugusto) [inline app/renderer/components/common/button.js]: YESSSSSSSS

File: app/renderer/components/common/modalOverlay.js [hunk 4]
```diff
@@ -108,11 +108,11 @@ class ModalOverlay extends ImmutableComponent {
...
   render () {
-    return <div className={cx({
-      [css(styles.dialog__wrapper_modal)]: true,
-      [css(styles.dialog__wrapper_last)]: this.state.last,
-      [css(styles.dialog__wrapper_transparentBackground)]: this.props.transparentBackground
-    })} role='alert'>
+    return <div className={css(
+      styles.dialog__wrapper_modal,
+      this.state.last && styles.dialog__wrapper_last,
+      this.props.transparentBackground && styles.dialog__wrapper_transparentBackground
```
Comment (cezaraugusto) [inline app/renderer/components/common/modalOverlay.js]: ❣️

File: app/renderer/components/download/downloadItem.js (hunk 2 above)
Comment (cezaraugusto) [reply app/renderer/components/download/downloadItem.js]: haha w00t 👨‍🎤 🎶

File: app/renderer/components/common/button.js (hunk 3 above)
Comment (luixxiul) [reply app/renderer/components/common/button.js]: There are normal `<button>` still left, which will be replaced with `BrowserButton` :-)

**Comment by luixxiul** [comment]
*Rebasing...*

File: app/renderer/components/navigation/navigator.js [hunk 5]
```diff
@@ -257,9 +263,24 @@ const styles = StyleSheet.create({
...
   braveMenuContainer: {
     position: 'relative'
   },
+
+  braveMenuContainer__braveShield_down: {
+    filter: 'grayscale(100%)'
+  },
+
+  braveMenuContainer__braveShield_disabled: {
+    filter: 'grayscale(100%)',
```
Comment (cezaraugusto) [inline app/renderer/components/navigation/navigator.js]: grayscale ftw

Comment (bsclifton) [reply app/renderer/components/navigation/navigator.js]: @luixxiul does this also fix https://github.com/brave/browser-laptop/issues/9696?

Comment (luixxiul) [reply app/renderer/components/navigation/navigator.js]: I don't think so as this does not change the color of the number block.

Comment (luixxiul) [reply app/renderer/components/navigation/navigator.js]: I have a fix for that issue. Is it fine to commit it to this PR?

screenshot 2017-09-22 16 20 15

Comment (bsclifton) [reply app/renderer/components/navigation/navigator.js]: Sure- add it in 😄

Comment (luixxiul) [reply app/renderer/components/navigation/navigator.js]: Added :-) https://github.com/brave/browser-laptop/pull/10951/commits/8615b5fb832965d72cb4072ecc420dc47fb04d9e

**Comment by cezaraugusto** [comment]
keen to review this but delaying to after 0.19.x, given it has some changes ledger-related. Setting as `PR/blocked` for now to keep track

**Comment by luixxiul** [comment]
Until 0.19 is released, I'm going to add other commits to normalize the buttons! :-)

**Comment by luixxiul** [comment]
The 4th commit requires https://github.com/brave/browser-laptop/pull/10789

**Comment by luixxiul** [comment]
This blocks #10881

**Comment by luixxiul** [comment]
It is a known issue and https://github.com/brave/browser-laptop/pull/10881/commits/70fae758fb9a5dd2553dce722a3ec6f8a17f14b2 of #10881 fixes that.

**Comment by luixxiul** [comment]
I'm going to open another issue for that bug and will create a PR with that commit.

**Comment by cezaraugusto** [comment]
is this font the right one? I think there are some efforts from design to change the actual font but not following so cc @petemill 

screen shot 2017-12-19 at 1 09 06 pm

**Comment by luixxiul** [comment]
I haven't checked the source code for a while, so let me close once to try to find a better fix, thanks.

**Pull request closed without merging**